dhis2-extraction/dhis2extract.py

```diff
@@ -485,18 +476,43 @@ def data_value_sets(
         if start_date and end_date:
             params["startDate"] = start_date
             params["endDate"] = end_date
-        if org_units:
-            params["orgUnit"] = org_units
         if org_unit_groups:
             params["orgUnitGroup"] = org_unit_groups
         if data_element_groups:
             params["dataElementGroup"] = data_element_groups
         if attribute_option_combos:
             params["attributeOptionCombo"] = attribute_option_combos
 
+        # The dataValueSets endpoint does not support the "ou:LEVEL-n" dimension
+        # parameter. As an alternative, we request data for the parent org units
+        # and include data for the children org units in the response.
+        # NB: org_units parameter still takes precedence.
+        if not org_units and org_unit_levels:
+            return self._data_value_sets_for_level(params, org_unit_levels)
+
+        return self._data_value_sets_regular(params)
+
+    def _data_value_sets_for_level(self, params, org_unit_levels):
+        org_units = []
+        for lvl in org_unit_levels:
+            org_units.append(self.org_units_per_lvl(lvl - 1))
+
+        params["include_childrens"] = True
+        df = pd.DataFrame()
+        for i in range(len(org_units), 50):
```

Comment:
I would put the chunk size as a parameter event if we don't expose it in the CLI yet